Automattic/jetpack
# Gutenframe: Show message if user is logged out

Fixes https://github.com/Automattic/wp-calypso/issues/32046.
Background: p3fqKv-6Hh-p2.

### Changes proposed in this Pull Request:

When the user is not logged into the Jetpack site, the WordPress.com iframed block editor won't load because the iframe redirects to the login URL which cannot be embedded in an iframe (since it causes a `sameorigin` error screen). 

<img width="816" alt="Screen Shot 2019-04-23 at 11 25 44 AM" src="https://user-images.githubusercontent.com/349751/56607121-9f288280-65bc-11e9-8add-61dc28dd4a71.png">

This PR aborts the default workflow with a `wp_die` command that displays a help message to the user explaining we need them to be logged in. The message is only displayed in a iframed block editor context (current page is `post.php` or `post-new.php` and the param `frame-nonce` is present).

<img width="856" alt="Screen Shot 2019-04-24 at 15 47 30" src="https://user-images.githubusercontent.com/1233880/56638633-9d88b880-66a9-11e9-846c-3922074a75c6.png">

#### Testing instructions:
* Make sure you're logged out the Jetpack site.
* Start Calypso locally using the `update/me-sites-version` branch (https://github.com/Automattic/wp-calypso/pull/32527).
* Load `/block-editor/post/:yourJetpackDomain`.
* Verify you see the custom error message.
* Load any other page of the Jetpack site (i.e. `://yourJetpackDomain/wp-admin`) and confirm you don't see that message and you're redirected to the login page.
* Login into the Jetpack site.
* Reload `/block-editor/post/:yourJetpackDomain`.
* Check the iframed block editor loads successfully now.

#### Proposed changelog entry for your changes:
I guess it's not needed?
